test/e2e/specs/specs-playwright/wp-signup__paid.ts

```diff
@@ -0,0 +1,181 @@
+	PaymentDetails,
+} from '@automattic/calypso-e2e';
+import { Page } from 'playwright';
+
+describe( DataHelper.createSuiteTitle( 'Signup: WordPress.com Paid' ), function () {
+	const inboxId = DataHelper.config.get( 'inviteInboxId' ) as string;
+	const username = `e2eflowtestingeditor${ DataHelper.getTimestamp() }`;
+	const email = DataHelper.getTestEmailAddress( {
+		inboxId: inboxId,
+		prefix: username,
+	} );
+	const signupPassword = DataHelper.config.get( 'passwordForNewTestSignUps' ) as string;
+	const blogName = DataHelper.getBlogName();
+
+	let page: Page;
+
+	setupHooks( ( args ) => {
+		page = args.page;
+	} );
+
+	describe( 'Signup via /start', function () {
+		const targetDomain = `${ blogName }.com`;
+
+		let cartCheckoutPage: CartCheckoutPage;
+
+		it( 'Navigate to Signup page', async function () {
+			const loginPage = new LoginPage( page );
+			await loginPage.clickSignup();
+		} );
+
+		it( 'Set store cookie', async function () {
+			await BrowserManager.setStoreCookie( page, { currency: 'GBP' } );
+		} );
+
+		it( 'Sign up as new user', async function () {
+			const userSignupPage = new UserSignupPage( page );
+			await userSignupPage.signup( email, username, signupPassword );
+		} );
+
+		it( 'Select a .com domain', async function () {
+			const domainSearchComponent = new DomainSearchComponent( page );
+			await domainSearchComponent.search( blogName );
+			await domainSearchComponent.selectDomain( targetDomain );
+		} );
+
+		it( 'Select WordPress.com Personal plan', async function () {
+			const signupPickPlanPage = new SignupPickPlanPage( page );
+			await signupPickPlanPage.selectPlan( 'Personal' );
+		} );
+
+		it( 'See secure payment', async function () {
+			cartCheckoutPage = new CartCheckoutPage( page );
+			await cartCheckoutPage.validateCartItem( 'WordPress.com Personal' );
+			// await cartCheckoutPage.validateCartItem( targetDomain );
+		} );
+
+		it( 'Prices are shown with expected currency symbol', async function () {
+			const buttonText = await cartCheckoutPage.getPaymentButtonText();
+			await buttonText.startsWith( '£' );
+		} );
+
+		it( 'Remove domain purchase from cart', async function () {
+			await cartCheckoutPage.removeCartItem( targetDomain );
+		} );
+
+		it( 'Apply coupon and validate resulting purchase amount', async function () {
+			const originalAmount = await cartCheckoutPage.getCheckoutTotalAmount();
+			await cartCheckoutPage.enterCouponCode();
+			const newAmount = await cartCheckoutPage.getCheckoutTotalAmount();
+
+			expect( newAmount ).toBeLessThan( originalAmount );
+			const expectedAmount = originalAmount * 0.99;
+
+			// Some currencies do not typically have decimal places.
+			// eg. USD would commonly have 2 decimal places, e.g. 12.34.
+			// In JPY or TWD there will be no decimal digits.
+			// Drop decimals so that the result won't be affected by the currency variation.
+			expect( Math.floor( newAmount ) ).toStrictEqual( Math.floor( expectedAmount ) );
```

Comment:
I actually feel like while it's good to handle this variability in the library, we don't need to do handle this here, and can just tailor this test case to whatever we expect for GBP. The test spec itself inevitably has to be coupled to a given currency, and I think that's okay!

Reply:
Could you clarify what you mean?

Instead of performing the math operation here, move it into the library?